shlr/gdb/src/messages.c

```diff
@@ -45,8 +45,158 @@ int handle_cmd(libgdbr_t* g) {
 	return send_ack (g);
 }
 
-int handle_connect(libgdbr_t* g) {
+int handle_qStatus(libgdbr_t* g) {
+	char *tok = NULL;
+	tok = strtok (g->data, ";");
+	// TODO: We do not yet handle the case where a trace is already running
+	if (strncmp (tok, "T0", 2)) {
+		send_ack (g);
+		return -1;
+	}
+	// Ensure that trace was never run
+	while (tok != NULL) {
+		if (!strncmp (tok, "tnotrun:0", 9)) {
+			return send_ack (g);
+		}
+	    tok = strtok (NULL, ";");
+	}
+	send_ack (g);
+	return -1;
+}
+
+int handle_qC(libgdbr_t* g) {
+	char *t1, *t2;
+	// We get process and thread ID
+	if (strncmp (g->data, "QC", 2)) {
+		send_ack (g);
+		return -1;
+	}
+	t2 = g->data + 2;
+	if ((t1 = strchr (g->data, 'p'))) {
+		if (!(t2 = strchr (g->data, '.'))) {
+			send_ack (g);
+			return -1;
+		} else {
+			t1++;
+			unpack_hex (t1, t2 - t1, (char*) &g->pid);
+			t2++;
+		}
+	}
+	unpack_hex (t2, strlen (t2), (char*) &g->tid);
+	return send_ack (g);
+}
+
+int handle_execFileRead (libgdbr_t* g) {
+    if (g->data[0] == 'E') {
+	send_ack (g);
+	return -1;
+    }
+    if (!g->data[1]) {
+	// We're supposed to get filename too
+	send_ack (g);
+	return -1;
+    }
+    g->exec_file_name = strdup (g->data + 1);
+    return send_ack (g);
+}
+
+int handle_qSupported(libgdbr_t* g) {
 	// TODO handle the message correct and set all infos like packetsize, thread stuff and features
+	char *tok = NULL;
+	tok = strtok (g->data, ";");
+	while (tok != NULL) {
+		if (!strncmp (tok, "PacketSize=", 11)) {
+			char temp_buf[20] = { 0 };
+			temp_buf[0] = '0';
+			temp_buf[1] = 'x';
+			snprintf (temp_buf + 2, 16, "%s", tok + 11);
+			g->stub_features.pkt_sz = strtoul (temp_buf, NULL, 16);
+		} else if (!strncmp (tok, "qXfer:", 6)) {
+			if (!tok[6]) {
+				tok = strtok(NULL, ";");
+				continue;
+			}
+			char *p = tok + 6;
+			if (!strncmp (p, "btrace:read", 11)) {
```

Comment:
`r_str_startswith` was giving incorrect results. Since it is `bool`, generated assembly is `sete al`. But the `if()` checks `test eax, eax`. Changing return type to `int` with next commit